File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/java/io/trino/tpcds/row/InventoryRow.java
package io.trino.tpcds.row;

    import java.util.List;

    import org.apache.kyuubi.spark.connector.tpcds.row.KyuubiTPCDSTableRowWithNulls;

    import static com.google.common.collect.Lists.newArrayList;
    import static io.trino.tpcds.generator.InventoryGeneratorColumn.INV_DATE_SK;
    import static io.trino.tpcds.generator.InventoryGeneratorColumn.INV_ITEM_SK;
    import static io.trino.tpcds.generator.InventoryGeneratorColumn.INV_QUANTITY_ON_HAND;
    import static io.trino.tpcds.generator.InventoryGeneratorColumn.INV_WAREHOUSE_SK;

public class InventoryRow
    extends KyuubiTPCDSTableRowWithNulls {
  private final long invDateSk;
  private final long invItemSk;
  private final long invWarehouseSk;
  private final int invQuantityOnHand;

  public InventoryRow(long nullBitMap, long invDateSk, long invItemSk, long invWarehouseSk,
      int invQuantityOnHand) {
    super(nullBitMap, INV_DATE_SK);
    this.invDateSk = invDateSk;
    this.invItemSk = invItemSk;
    this.invWarehouseSk = invWarehouseSk;
    this.invQuantityOnHand = invQuantityOnHand;
  }

  public long getInvDateSk() {
    return invDateSk;
  }

  public long getInvItemSk() {
    return invItemSk;
  }

  public long getInvWarehouseSk() {
    return invWarehouseSk;
  }

  public int getInvQuantityOnHand() {
    return invQuantityOnHand;
  }

  @Override public Object[] values() {
    return new Object[] {
        getOrNullForKey(invDateSk, INV_DATE_SK),
        getOrNullForKey(invItemSk, INV_ITEM_SK),
        getOrNullForKey(invWarehouseSk, INV_WAREHOUSE_SK),
        getOrNull(invQuantityOnHand, INV_QUANTITY_ON_HAND)
    };
  }
}

File: extensions/spark/kyuubi-spark-connector-tpcds/src/main/scala/io/trino/tpcds/row/InventoryRow.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.trino.tpcds.row

import java.util.{List => JList}

import io.trino.tpcds.generator.InventoryGeneratorColumn._
import org.apache.spark.sql.catalyst.InternalRow
import org.apache.spark.sql.catalyst.expressions.GenericInternalRow

class InventoryRow(
    nullBitMap: Long,
    private val invDateSk: Long,
    private val invItemSk: Long,
    private val invWarehouseSk: Long,
    private val invQuantityOnHand: Int) extends TableRowWithNulls(nullBitMap, INV_DATE_SK)
  with KyuubiTableRowWithNulls {

  override val nullBitMapInternal = nullBitMap
  override val firstColumnInternal = INV_DATE_SK
  def getInvDateSk: Long = invDateSk
  def getInvItemSk: Long = invItemSk
  def getInvWarehouseSk: Long = invWarehouseSk
  def getInvQuantityOnHand: Int = invQuantityOnHand

  override def getValues: JList[String] = throw new UnsupportedOperationException()

  def internalRow: InternalRow =
    new GenericInternalRow(
      Array(
        getLongOrNull(invDateSk, INV_DATE_SK),
        getLongOrNull(invItemSk, INV_ITEM_SK),
        getLongOrNull(invWarehouseSk, INV_WAREHOUSE_SK),
        getIntOrNull(invQuantityOnHand, INV_QUANTITY_ON_HAND)))
}
